projectional/src/main/java/jetbrains/jetpad/projectional/cell/BaseProjectionalSynchronizer.java

```diff
@@ -275,15 +276,29 @@ public void addMapperProcessor(MapperProcessor<SourceItemT, Cell> processor) {
 
   @Override
   public void setOnLastItemDeleted(Runnable action) {
+    if (!myPlaceholderEnabled && action == null) {
+      throw new IllegalArgumentException("Can't set the default onLastItemDeleted action when the placeholder is disabled");
+    }
     myOnLastItemDeleted = action;
   }
 
   @Override
   public void setPlaceholderText(String text) {
+    if (!myPlaceholderEnabled) {
+      throw new IllegalStateException("The placeholder is disabled");
+    }
     myPlaceholderText = text;
   }
 
   @Override
+  public void disablePlaceholder() {
+    if (myOnLastItemDeleted == null) {
+      throw new IllegalStateException("Please provide onLastItemDeleted action first by calling setOnLastItemDeleted()");
```

Comment:
At least we should mention something about setting the focus then in the exception message.